### ISSUE TITLE
3 Features that might help?

### ISSUE THREAD

**Issue by thedunebread** [KomodoPlatform/atomicDEX-Pro#481]
Hi, let me start of by saying I’m not a dev so I can’t judge whether any of these ideas are helpful or even possible, but hopefully it will be of some use. 

1) When tx fees are high, ive seen a few community members question this or not understand that blockchain tx fees need to be paid. An example is if someone wants to buy say 1000kmd and it takes up 5 orders on the order book. By paying for 5 btc tx or eth when fees are high, it may seem abnormally high. 

I suggest having a hoverable ? Next to fee that breaks it down. Ie explanation of where it comes from, then maybe listen tx numbers or breaking it down (ie 5 btc tx @ 0.00001 fee). This way people know it is not a fee set by atomicdex but from the orders. Which also brings me to another point where if minimum order is 0.00777, someone could hypothetically make many orders 1 sat above each other in a way where anyone wanting to buy an order above would have to pay big fees (say $10) on all of them before reaching any sizeable amount? 

Would a feature in the future maybe be able to skip an order temporarily if the fee is higher than a set % of the order size? If I wanted to buy 1000kmd and there’s 3 orders, 400kmd @ 0.001, 1kmd @ 0.011 and 600kmd & 0.012, the ability to skip the middle order or have it automatically disclosed depending on the fees?

2) I saw a mention on liquidity pools and it got me thinking. Could we possibly have “special” orders, which are different to normal ones (say highlighted yellow) in which a market maker can plug into an existing liquidity pool or exchange. If someone wants to buy or sell a token, the “special” order is permission to trade from one of these special orders. 

I could say set the bot to arb any trade on any token without having to actually have the funds at that moment. Ie set an order to sell *user set/unlimited” kmd @ 1usdt. 

I don’t actually hold the kmd but I have my account active on the pool of liquidity. If someone accepts my order, they lock the funds in for a specified max time, which then sends a message to the seller to go ahead and buy on the exchange and send to atomicdex to complete the trade. 

Doing so means we could provide liquidity at much higher rates, and across every pair (as long as the seller presumes it is profitable) without actually having to physically own the coins. In essence an arbitration tool. 

I know atomicdex is focused solely on atomic swaps but this would still be an AS. It would just let a lot more people in and allow them to provide liquidity from any pair and any exchange and always be profiting in BTC or USDT. 

It would look something like 
Seller puts up orders on all pairs available on their binance account or currency.com account 

Atomic dex shows seller across many pairs in massive volume @ seller set price 

The order is highlighted in yellow dictating it may take more time to complete than a normal swap

Buyer takes offer and deposits coins for trade which are locked

Atomicdex sends notification to sellers LP that trade is initiated, they then automatically buy on exchange for set price in BTC USD or whatever main currency

Funds are sent out of exchange with api to atomicdex 

Trade is completed

im not sure if this is something workable, but the main reason for me not providing liquidity is profit as well as having to physically buy many tokens and receive back tokens I may not want as profit. 

If we could set orders easily like that, using exchange price and have it automated, I can see millions of volume coming in. Ie if I sat on 10btc I could provide 10btc sell volume on every coin, without having to physically own any of them. 

This would improve order book liquidity massively and make it easier for everyone to take advantage of arbing exchanges and lowering the difference in price between cex and dex.

**Comment by tonymorony** [comment]
Hi. Thank you for the issue opening and thoughts sharing!

1) 

```
When tx fees are high, ive seen a few community members question this or not understand that blockchain tx fees need to be paid. An example is if someone wants to buy say 1000kmd and it takes up 5 orders on the order book. By paying for 5 btc tx or eth when fees are high, it may seem abnormally high.
```

Right now protocol not matching >1 order automatically so user will see information about fees on each trade confirmation. If there is many orders with minor price difference from some "juicy" order it's possible to explicitly select this huge order and match with it

```
Would a feature in the future maybe be able to skip an order temporarily if the fee is higher than a set % of the order size? If I wanted to buy 1000kmd and there’s 3 orders, 400kmd @ 0.001, 1kmd @ 0.011 and 600kmd & 0.012, the ability to skip the middle order or have it automatically disclosed depending on the fees?
```

at the moment it works like a: if you place 1000kmd order but there are 400, 1 and 600 depth orders it will not match any of them but will place your maker order instead. To sell 1000 you should execute two trades: for 400 kmd @0.001 and for 600 kmd @0.012

2) Yeah thats great to have protocol mechanisms to combine funds for rollover in some automated strategy. I think we'll work on things in such direction next year.

With CEXes it's quite tricky matter: atomic swaps taking around 1 hour, deposit of for example KMD to Binance taking 100 minutes in avg. at the moment what making arbitrage and trading strategies with CEXes involment quite complicated to implement (not impossible however)

I also thought this one might increase interest in more maker orders placing (=liqudity): https://github.com/KomodoPlatform/atomicDEX-API/issues/714

Also this one: https://github.com/KomodoPlatform/atomicDEX-API/issues/715 might make BTC swaps (especially when few orders needs to be matched to fill full taker volume such as you worrying about in 1) ) much cheaper by avoiding BTC txfees for fee tx paying.

**Comment by thedunebread** [comment]
I agree arbing in a general sense would be hard. However speaking from a
liquidity only standpoint, having special orders set as I mentioned should
be doable, and as long as the buyer or seller understands that it will take
longer than necessary, I don’t see an issue with it.

Normal arbitrage is obviously very time sensitive and can have issues with
selling in time to make the profit.

However in this case, it will always turn out profitable as it is not
trying to both buy and sell, but simply providing the liquidity.

Let’s say for example person a is the market maker, and person b is the
buyer.

Komodo price is currently $1

Person a has 1btc and sets up his “special” sell orders.

He places a special order for 1btc worth of kmd @ $1.02 / 1btc worth of
every pairing @ 2% higher than the exchange he is plugged into.

Someone who was going to buy Komodo sees the order. There are no cheaper
orders, and he is fine with waiting as he wants to use dex for his own
reasons or because it is the cheapest he has access to.

Person b locks in 1btc to purchase kmd for himself. He is happy with the
deal.

Person a has 1btc on the exchange already, so because person b has locked
in the 1btc, his market maker automatically buys the kmd on the exchange
(which costs him 2% under 1btc) and it is sent to atomicdex to claim the
1btc.

I think negative fees and competitions will increase liquidity slightly,
but for big amounts I don’t think it will do much in the long term. If I
wanted purely btc profits, that is not currently possible as I would have
to hold multiple coins to provide the pairs.

Using the above system, it seems that it can increase liquidity on all
available pairs at once, in large numbers and pretty much gap the whole
liquidity from cex to dex without having to have multiple coins in the
first place (which is what is keeping me personally from running one).

I’m not sure if I explained it properly but let me know what you think. I
believe having large sell orders available across all main pairs will help
bring atomicdex useage up.
